### PR TITLE
feat(dbt-fal): make fal-scripts-path value available as a path to import for models

### PR DESCRIPTION
**Pull request by chamini2**

### CODE REVIEW

File: adapter/src/dbt/adapters/fal_experimental/adapter.py [hunk 1]
```diff
@@ -71,8 +76,32 @@ def run_in_environment_with_adapter(
...
+    try:
+        yield
+    finally:
+        sys.path.remove(path)
+
+def _get_fal_scripts_path(config: RuntimeConfig):
+    import pathlib
+    cwp = pathlib.Path.cwd()
+
+    fal_scripts_path = config.project_root
```
Comment (mederka) [inline adapter/src/dbt/adapters/fal_experimental/adapter.py:99]: Is `project_root` a sane default for script paths?

File: adapter/src/dbt/adapters/fal_experimental/adapter.py [hunk 2]
```diff
@@ -71,8 +76,32 @@ def run_in_environment_with_adapter(
...
+
+    if hasattr(config, 'vars'):
+        fal_scripts_path: str = config.vars.to_dict().get(FAL_SCRIPTS_PATH_VAR_NAME, fal_scripts_path)  # type: ignore
+
+    if hasattr(config, 'cli_vars'):
+        fal_scripts_path = config.cli_vars.get(FAL_SCRIPTS_PATH_VAR_NAME, fal_scripts_path)
+
+    return cwp / fal_scripts_path
```
Comment (mederka) [inline adapter/src/dbt/adapters/fal_experimental/adapter.py:107]: ATL: What does this path look like if fal_scripts_path is the project root directory?

File: adapter/src/dbt/adapters/fal_experimental/adapter.py [hunk 3]
```diff
@@ -71,8 +76,32 @@ def run_in_environment_with_adapter(
     else:
         deps = get_default_pip_dependencies()
         stage = VirtualPythonEnvironment(deps)
+        fal_scripts_path = _get_fal_scripts_path(config)
 
-        with PythonIPC(environment, environment.create(), extra_inheritance_paths=[stage.create()]) as connection:
+        with PythonIPC(environment, environment.create(), extra_inheritance_paths=[fal_scripts_path, stage.create()]) as connection:
```
Comment (mederka) [inline adapter/src/dbt/adapters/fal_experimental/adapter.py:81]: 🙌

**Review by mederka — COMMENTED** [review]
Great job, @chamini2! I just got a couple of comments.

File: adapter/src/dbt/adapters/fal_experimental/adapter.py (hunk 1 above)
Comment (chamini2) [reply adapter/src/dbt/adapters/fal_experimental/adapter.py:99]: This is what we had for fal cli and what we have documented for it. But it is a good question! We could just NOT include anything if not specified.

File: adapter/src/dbt/adapters/fal_experimental/adapter.py [hunk 4]
```diff
@@ -71,8 +76,33 @@ def run_in_environment_with_adapter(
...
+    import sys
+    sys.path.append(path)
+    try:
+        yield
+    finally:
+        sys.path.remove(path)
+
+def _get_fal_scripts_path(config: RuntimeConfig):
+    import pathlib
+    cwp = pathlib.Path.cwd()
```
Comment (isidentical) [inline adapter/src/dbt/adapters/fal_experimental/adapter.py:97]: ATL: shouldn't this be project root instead of the cwd?

**Review by isidentical — APPROVED** [review]

**Pull request merged**